src/docdb/nodes.ts

```diff
@@ -178,4 +180,24 @@ export class DocDBDocumentNode implements IDocumentNode {
 
 		return this._data;
 	}
-}
+
+  getPartitionKeyValue(): string {
+		const partitionKey = this.collection.partitionKey;
+		if (!partitionKey) {
+			return null;
+		}
+		const fields = partitionKey.paths[0].split('/');
+		if (fields[0] === '') {
+			fields.shift();
+		}
+		let value;
+		for (let field of fields) {
+			value = value ? value[field] : this.data[field];
+			if (!value) {
```

Comment:
Looping is similar to the LoadMore() method. 
Break if at any point, the field is null/undefined (implying that the partitionKey path doesn't exist in that document).